### ISSUE TITLE
Create degraded pool

### ISSUE THREAD

**Issue by ChristophSchmidpeter** [openzfs/zfs#16367]
### Describe the feature would like to see added to OpenZFS

I would like the ability to create a pool even if I don't have all physical disks at the time of pool creation.

Imagine you want to create a raidz pool with three disks. Normally, you would create the pool e.g. via
`zpool create mypool raidz /dev/sda /dev/sdb /dev/sdc`

I would like to be able to create the pool even if I have only 2 disks available right now. The pool would be in a degraded state and at a later point in time I could resilver the third disk.

### How will this feature improve OpenZFS?
Better user flexibility

**Comment by gmelikov** [comment]
You can just create sparsed file via `truncate` and use it as one of disks. Just `zpool offline` it after pool creation.

**Comment by snajpa** [comment]
I agree this would be useful, a lot. We've had situations where we'd almost be able to build out a new array in a system, if we could temporarily skip all of the drives with redundant data. I won't juggle around with workarounds, to me this is currently unsupported, I can see why people could find it useful to have that supported directly.

**Comment by Haravikk** [comment]
> You can just create sparsed file via truncate and use it as one of disks. Just zpool offline it after pool creation.

This is what I've done in the past and it works fine, it is a bit of a pain in the ass though; it would be nice if I could just put `empty` or such in the `zpool create` command to indicate "to be added", perhaps with confirmation that this will create the pool in a degraded state (and by how much).

It's a pretty niche feature though, but it would eliminate the need for sparse file juggling to provide fake drives and then remove them.
